### DISCUSSION TITLE
Possible websocket issue with remote icon loading?

### DISCUSSION THREAD

**Discussion by echto** [dani-garcia/vaultwarden#2613] [Q&A]
I am having trouble narrowing down this issue. 

Arch OS package : vaultwarden 1.25.0
Reverse proxy : Nixos nginx/1.16.1
Database : Nixos postgres (PostgreSQL) 14.4

I keep losing the SSH connection (send disconnect: Broken pipe) to an Arch linux VM that runs vaultwarden.  Rebooting the VM restores network connectivity.  Could this issue be related to websockets and when the internal icon download service cannot find an icon, i.e. '404 Not Found' ?  I am assuming the default icon downloading service is internal. No icon service is defined in vaultwarden.env.  Please see the following pastebin link for log.  Is it normal to see a burst of 'parity_ws::handler][DEBUG] Handler received' messages?

https://pastebin.com/vYRjEAmE

**Comment by BlackDex** [comment]
Could you please try the `testing` tagged images please, and see if that solves it? Those versions do not use `parity_ws` anymore.

**Comment by echto** [comment]
I will try that.

**Comment by echto** [comment]
Issue resolved.  Please close.

**Reply by echto** [reply]
I spoke too soon.  The issue is still happening.

**Comment by BlackDex** [comment]
I really think it's not an actual issue with Vaultwarden, but with a reverse proxy config. And, what logs do you get right now? Is it the same. What do the host logs say? So the docker runtime logs, or other host logs. Or the reverse proxy logs etc.. etc...

**Comment by echto** [comment]
I don't know what is it.  I doubt is it the reverse proxy.  I have other services using a reverse proxy without issue.  There is nothing in the logs.  I have tried running in a docker container and stand alone.  Host logs don't show anything.  Believe me, I have tried to find something.  The only thing left for me to try is bitwarden itself and see of the same issue occurs.

**Comment by echto** [comment]
I believe I may have discovered the issue.  I just happened to now witness a bot scan which killed my ssh connection.  I will post more info later.

**Reply by BlackDex** [reply]
Well, that is something the reverse proxy should be able to handle, and close connects. This looks like all connects keep open and causes all connections to be depleted.

**Comment by echto** [comment]
Jul 15 21:20:58 pass systemd-coredump[925]: [🡕] Process 785 (vaultwarden) of user 0 dumped core.
                                           
                                            Module /root/vaultwarden with build-id f7fc3c57f89555b9463662899df9ebb04bbe9a1e
                                            Module /root/vaultwarden with build-id f7fc3c57f89555b9463662899df9ebb04bbe9a1e
                                            Module /usr/lib/libc.so.6 with build-id 60df1df31f02a7b23da83e8ef923359885b81492
                                            Module linux-vdso.so.1 with build-id f7714272211882ce0c201f61fdde178e0a74d697
                                            Module libsasl2.so.3 with build-id 0c10b132405d1c16805173607ae5496c9bc84ba7
                                            Module liblber.so.2 with build-id c3d8f0b19222aef1692d78535f3e795760bf07a1
                                            Module libresolv.so.2 with build-id 89a368a6ad1b392d126a2a5beb9c2f61ade00279
                                            Module libkeyutils.so.1 with build-id ac405ddd17be10ce538da3211415ee50c8f8df79
                                            Module libkrb5support.so.0 with build-id 15f223925ef59dee4379ebbc0fcd14eda9ba81a2
                                            Module libcom_err.so.2 with build-id 3360a28740ffbbd5a5c0c21d09072445908707e5
                                            Module libk5crypto.so.3 with build-id cc77a742cb62447a53d98285b41558b8acd92866
                                            Module libkrb5.so.3 with build-id 371cc767dacb17cb42c9c44b88eebbed5ee9a756
                                            Module libz.so.1 with build-id fefe3219a96d682ec98fcfb78866b8594298b5a2
                                            Module libldap.so.2 with build-id c9be4f0aa087ad90d69a828c34413da4f1ee3ea3
                                            Module libgssapi_krb5.so.2 with build-id 292f1ce32161c0ecc4a287bc8494d5d7c420a03f
                                            Module ld-linux-x86-64.so.2 with build-id 0effd0e43efa4468d3c31871c93af0b7f3005673
                                            Module libc.so.6 with build-id 60df1df31f02a7b23da83e8ef923359885b81492
                                            Module libdl.so.2 with build-id 94198b268228074fa9f405bbedbbae94112593ed
                                            Module libm.so.6 with build-id 1b7296ef9fd806e47060788389293c824b09ad72
                                            Module libpthread.so.0 with build-id 95ae4f30a6f12ccbff645d30f8e1a3ee23ec7d36
                                            Module libgcc_s.so.1 with build-id 0e3de903950e35ae59a5de8c00b1817a4a71ca01
                                            Module libmariadb.so.3 with build-id 8bae3e7345a5e20bb827facc69847ea135ffd067
                                            Module libpq.so.5 with build-id 2287b499054ddd7e3665b8cbc6d22d7646e2dedd
                                            Module libcrypto.so.1.1 with build-id 7981ea3d69f3c28e46ee312a815af96eab93775c
                                            Module libssl.so.1.1 with build-id e6b1f97a5b60b4248c49dfc5b11f53f281b507d0
                                            Stack trace of thread 799:
                                            #0  0x00007f1c3768e36c n/a (libc.so.6 + 0x8e36c)
                                            #1  0x00007f1c3763e838 raise (libc.so.6 + 0x3e838)
                                            #2  0x00007f1c37628535 abort (libc.so.6 + 0x28535)
                                            #3  0x000055e845805617 n/a (/root/vaultwarden + 0x7c3617)
                                            ELF object binary architecture: AMD x86-64

Jul 15 21:20:58 pass systemd[1]: systemd-coredump@0-924-0.service: Deactivated successfully.

**Comment by BlackDex** [comment]
Well, i can't really extract anything useful from that, except the fact that you are using Vaultwarden via a systemd environment.

I would suggest to read the following https://github.com/dani-garcia/vaultwarden/wiki/Setup-as-a-systemd-service

And also look for LimitNPROC in issues and discussions https://github.com/dani-garcia/vaultwarden/search?q=LimitNPROC&type=discussions 

There are a few know weird behaviors with some systemd configs for some reason. Especially within lxd/lxc containers, and maybe also other virtual environments.

**Comment by echto** [comment]
I do not know what is going on with vaultwarden.  I disabled vaultwarden and ran a separate ngnix webserver on the same VM hosting a couple mp3 files that I continually downloaded over the past two days, local and remote, and not one network issue occurred.  I start up vaultwarden and 12 hours later I loose the network connection again.  It's quite clear it isn't my reverse proxy or the VM.

**Reply by BlackDex** [reply]
We need more info, trace logs, network or  something. This is a vague issue nobody else reported.

Also, running an nginx with file downloading is not really comparable.

Could you try to run Vaultwarden as a docker container and see what happens?

**Reply by BlackDex** [reply]
Or, since you think it has something to do with either websockets or icon downloading, which btw have no connection with each other, you could try to disable the websockets or icon downloading and see what happens then.

https://github.com/dani-garcia/vaultwarden/blob/f1a67663d19013271ebd49f7cd6056f59f6dc3f7/.env.template#L68

https://github.com/dani-garcia/vaultwarden/blob/f1a67663d19013271ebd49f7cd6056f59f6dc3f7/.env.template#L171

**Reply by echto** [reply]
update. i switched from systemd to openrc and I have not had an issue since.

**Reply by BlackDex** [reply]
Yea, thats why i mentioned to search for systemd. Since we know some configurations especially with container platforms like lxc/lxd/machined could cause strange issues in combination with systemd.

It either runs out of allowed processes or some other restrictions.
